extras/ioc_hash_scan.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+#
+##Add the hashes to the blob at src/ioc_scan/ioc_scanner.py
+#
+#
+#  The filename specified in the first argument
+# (instance-list-file) should contain a list of instance id strings, one per line.
+#
+# Usage: ./ioc_hash_scan.sh instance-list-file <AWS_PROFILE>
```

Comment:
I thought about grabbing `AWS_PROFILE` as an environmental profile - but I'm not sure how to do that well. Consider that there are likely a dozen or so possible `AWS_PROFILE`'s that someone may have - setting a single one as an environmental profile could create unnecessary pain and suffering.

I get this is likely the roughest part. I would like to simplify it.

Reply:
> Consider that there are likely a dozen or so possible `AWS_PROFILE`'s that someone may have - setting a single one as an environmental profile could create unnecessary pain and suffering.

It's trivial to call the script with only the profile and other env variables that you want, like so:
```console
AWS_SHARED_CREDENTIALS_FILE=~/.aws/my-creds AWS_REGION=us-east-1 AWS_PROFILE=my-profile ./ioc_hash_scan.sh i-0123456789abcdef
```

Reply:
See a2162cf552655c652e149bbd1611652eb3401170

Reply:
Typically, you don't include environment variables in example usage (e.g. [here](https://github.com/cisagov/ioc-scanner/blob/a2162cf552655c652e149bbd1611652eb3401170/extras/ioc_hash_scan.sh#L8) and [here](https://github.com/cisagov/ioc-scanner/blob/a2162cf552655c652e149bbd1611652eb3401170/extras/ioc_hash_scan.sh#L22)), so I suggest removing them from those lines.

Instead, I think you should mention in the [script's description comment](https://github.com/cisagov/ioc-scanner/blob/a2162cf552655c652e149bbd1611652eb3401170/extras/ioc_hash_scan.sh#L3-L6) that it expects certain environment variables to be set and be clear about which environment variables are needed.

Additionally, since those environment variables are required for this script to function properly, the script should check that they are indeed set and throw an error if they are not.

Reply:
I'm not sure how to test for the variables correctly. 
To test, I added this to the script, but did not pass the variable
`[[ -z "$AWS_PROFILE" ]] && { echo "AWS_PROFILE environmental variable is not set. Exiting"; exit 1; }`
But instead of the error output I expected, I got
`AWS_PROFILE: unbound variable`

Reply:
Much better, thanks!